app/features/quizzify/tools.py

```diff
@@ -102,61 +358,111 @@ def load(self) -> List[Document]:
+                    metadata = {"source": file_path}
                     doc = Document(page_content=page_content, metadata=metadata)
                     documents.append(doc)
 
+                elif file_type in ["ppt", "pptx"]:
+                    page_content = extract_text_from_pptx(file)
+                    metadata = {"source": file_path}
+                    doc = Document(page_content=page_content, metadata=metadata)
+                    documents.append(doc)
+
+                elif file_type == "csv":
+                    df = pd.read_csv(file)
+                    page_content = df.to_string()
+                    metadata = {"source": file_path}
+                    doc = Document(page_content=page_content, metadata=metadata)
+                    documents.append(doc)
+
+                else:
+                    raise ValueError(f"Unsupported file type: {file_type}")
+
+                
+
         return documents
 
 class URLLoader:
-    def __init__(self, file_loader=None, expected_file_type="pdf", verbose=False):
-        self.loader = file_loader or BytesFilePDFLoader
-        self.expected_file_type = expected_file_type
+    def __init__(self, file_loader=None, expected_file_types=None, verbose=False):
+        self.loader = file_loader or BytesFileLoader
+        self.expected_file_types = expected_file_types or ["pdf", "doc", "docx", "ppt", "pptx", "csv"]
         self.verbose = verbose
-    
-    def load(self, tool_files: list[ToolFile]) -> list[Document]:
+
+    def load(self, tool_files: List[ToolFile]) -> List[Document]:
         queued_files = []
-        
+        documents = []
+        any_success = False
+
         for tool_file in tool_files:
-            url = tool_file.url
-            response = requests.get(url)
-            
-            parsed_url = urlparse(url)
-            path = parsed_url.path
-            
-            if response.status_code == 200:
-                # Read file
-                file_content = BytesIO(response.content)
-                
-                # Check file type
-                file_type = path.split(".")[-1]
-                if file_type != self.expected_file_type: 
-                    raise ValueError(f"Expected file type: {self.expected_file_type}, but got: {file_type}")
-                
-                # Append to Queue
-                queued_files.append((file_content, file_type))
-                if self.verbose: logger.info(f"Successfully loaded file from {url}")
-                
-            else: logger.error(f"Request Failed to load file from {url}")
-            
-            # Pass Queue to the file loader
+            try:
+                url = tool_file.url
+                response = requests.get(url)
+                parsed_url = urlparse(url)                
+                path = parsed_url.path
+                                
+                if path.endswith(('.pdf', '.doc', '.docx', '.ppt', '.pptx', '.csv')):
```

Comment:
Must be an easier way to inject the list here rather than hardcoding it. You will find this not very scalable as it doesn't follow DRY (don't repeat yourself) principle

Reply:
You're right. I had refactored the file extension checking to use dynamic list. Please verify the changes